Lib/tarfile.py

```diff
@@ -2567,7 +2568,19 @@ def chmod(self, tarinfo, targetpath):
         if tarinfo.mode is None:
             return
         try:
-            os.chmod(targetpath, tarinfo.mode)
+            try:
+                os.chmod(targetpath, tarinfo.mode)
+            except OSError as e:
+                if not(hasattr(errno, "EFTYPE") and e.errno == errno.EFTYPE):
```

Comment:
Maybe only retry chmod() when it's needed? Or is this test redundant with checking for EFTYPE?

```suggestion
                if not(hasattr(errno, "EFTYPE") and e.errno == errno.EFTYPE and (tarinfo.mode & stat.S_ISVTX)):
```

Reply:
Yes, at least on FreeBSD. The documentation is very explicit about the condition causing the error:

> The effective user ID is not the super-user, the mode includes the sticky bit (S_ISVTX), and path does not refer to a directory.

I don't know of other systems with the same error, so the hasattr() check might be enough. But it might be a good idea to be defensive. It's not like it's an expensive check.

Reply:
Ok, no need to test for the bit in this case.